pydis_site/apps/api/tests/test_infractions.py

```diff
@@ -305,6 +312,167 @@ def test_returns_400_for_non_hidden_required_hidden_type(self):
             'hidden': [f'{data["type"]} infractions must be hidden.']
         })
 
+    def test_returns_400_for_active_infraction_of_type_that_cannot_be_active(self):
+        """Test if the API rejects active infractions for types that cannot be active."""
+        url = reverse('bot:infraction-list', host='api')
+        restricted_types = ('note', 'warning', 'kick')
+
+        for infraction_type in restricted_types:
+            with self.subTest(infraction_type=infraction_type):
+                invalid_infraction = {
+                    'user': self.user.id,
+                    'actor': self.user.id,
+                    'type': infraction_type,
+                    'reason': 'Take me on!',
+                    'hidden': True,
+                    'active': True,
+                    'expires_at': '2019-10-04T12:52:00+00:00'
```

Comment:
This test is a bit fragile in the sense that it relies on the active validation logic preceding the validation for allowing expiration or being hidden. But I think I can live with that.

Reply:
I can make the payload more dependent on the type, but it would mean a logically more complex test.

Reply:
Maybe not that much more complex. I'm wondering how it'd look if the expiration and hidden args could be put in the tuple alongside the types.

Reply:
Better, I think.